lib/Controller.js

```diff
@@ -190,18 +190,20 @@ module.exports = (function(){
                 return liveEntry.stop()
                     .then(function() {
                         logger.debug("[%s] Stopped entry. Removing it from list of handled entries", noLongerLiveEntry);
-                        delete that.handledEntries[noLongerLiveEntry];
                     })
```

Comment:
My fix was wrong, because the result is deletion of entry regardless of grace time. Code was reverted.